documentation/docs/developer_guide/contribution_flowchart.md

```diff
@@ -0,0 +1,162 @@
+---
+hide:
+  - toc
+---
+
+# Flowchart to contribute to CABLE's documentation
+
+```mermaid
+   %% Create a graph for the legend of the main graph
+   flowchart LR
+      uniq[action done <br> only once ever <br> at the start <br> to get the code locally]:::uniq;
+      sevterm[action done <br> several times per issue <br> in a terminal/text editor]:::term;
+      sevgit[action done <br> several times per issue <br> on GitHub]:::github;
+      onceterm([action done <br> once per issue <br> in a terminal/text editor]):::term;
+      oncegit([action done <br> once per issue <br> on GitHub]):::github;
+      question[\question with multiple outcomes/]:::question;
+
+      uniq --- sevterm --- sevgit --- question;
+      uniq --- onceterm --- oncegit --- question;
+
+      classDef default fill:#FFFDE7, stroke:#FFF59D;
+      classDef uniq fill:#D81B60, stroke:#880E4F, color:#FFFFFF;
+      classDef github fill:#388E3C,stroke:#1B5E20, color:#FFFFFF;
+      classDef term fill:#F44336, stroke:#B71C1C, color:#FFFFFF;
+      classDef question fill:#6D4C41, stroke:#3E2723, color:#FFFFFF; 
+;
+```
+
+```mermaid
+   flowchart TB
+
+      %% Define all the nodes first
+      clone[Clone repository]:::uniq ;
+      issue([Open an issue <br> to explain your work]):::github;
+      branch([Create a branch]):::github;
+      fetch([Fetch branch <br> to your local repository: <br> git fetch]):::term;
+      locbranch(["Create local branch: <br> git checkout &lt;branchname&gt;"]):::term;
+      edits[Make your edits]:::term;
+      commit["Commit your edits: <br> git commit -a -m &lt;message&gt;"]:::term;
+      push[Push your edits <br> git push]:::term;
+      first[\Is it first push?/]:::question;
+      pr([Create a pull request]):::github;
+      preview[\Is preview correct?/]:::question;
+      review[Request review]:::github;
+      reviewchange[\Review asks for code changes <br> or has comments?/]:::question;
+      revieweredits[\Reviewer has committed some edits? /]:::question;
```

Comment:
I'd make the reviewer committing edits a special case outside this main flow-chart. Maybe it is more common for you, but in general I don't think it happens that much.

Reply:
Currently, it is quite common. I agree it's usually not that often. But what might happen is to accept suggestions from the review on GitHub and then have to put some local edits as well. I'll make a special case.

Reply:
Or I don't make a special case and I just put to run `git pull` anyway? It doesn't hurt if things are the same and it's helpful if some commits were made on the PR either by the reviewer or by accepting suggestions.